app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.kt

```diff
@@ -4,19 +4,48 @@ import ar.com.wolox.android.example.utils.UserSession
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import javax.inject.Inject
 
+fun flattenSize(m: MutableMap<String, MutableList<String>>): Int {
+    var acc = 0
+    m.forEach { (s, mutableList) -> acc += mutableList.size }
+    return acc
+}
+
 class ExamplePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<ExampleView>() {
 
-    fun onLoginButtonClicked(user: String, color: String) {
-        userSession.username = user
-        view?.goToViewPager(color)
-    }
+    fun onLoginButtonClicked(email: String, password: String) {
+        var ErrorList: MutableMap<String, MutableList<String>> = mutableMapOf()
+
+        // TODO: How to avoid these lines ? I mean, how to create ErrorList[...] while inserting element in its list
+        ErrorList["email"] = mutableListOf()
+        ErrorList["password"] = mutableListOf()
 
-    fun onEmptyFieldsDetected() = null
+        // TODO: How can I make this lines "more" general ?
```

Comment:
What I recently thought (not in this case, but in a _more than two field validation_ scenario) is that you can disable login button until all the fields are ok. And the fields validations is made during user input (i.e. after user writes invalid@@mail in mail field, an alert should appear).

Reply:
Same, I think we should talk about this